upstairs/src/client.rs

```diff
@@ -2956,11 +2956,6 @@ pub(crate) fn validate_encrypted_read_response(
         }
     };
 
-    let Some(block_encryption_ctx) = &context.encryption_context else {
-        // this block context is missing an encryption context!
-        return Err(CrucibleError::DecryptionError);
-    };
-
     // We'll start with decryption, ignoring the hash; we're using authenticated
```

Comment:
this comment needs updating, and perhaps an additional line or two about how packet corruption currently causes decryption failure.

Reply:
Yup, done in 5ae6bd0137b9